python310Packages.numba: unstable-2023-08-02 -> unstable-2023-08-11

khaneliman opened the pull request:
## Description of changes
Just saw there were some updates while investigating another issue. 
Add meta.mainProgram 

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

khaneliman committed:
python310Packages.numba: unstable-2023-08-02 -> unstable-2023-08-11

FRidh commented:
See also #248316

Mic92 commented on pkgs/development/python-modules/numba/default.nix:
```diff
@@ -27,17 +27,16 @@
   inherit (cudaPackages) cudatoolkit;
 in buildPythonPackage rec {
-  # Using an untagged version, with numpy 1.25 support, when it's released
-  # also drop the versioneer patch in postPatch
```
we want to keep that hint, no?

doronbehar replied:
> we want to keep that hint, no?

I agree.

khaneliman replied:
Sorry must have gotten removed when I resolved merge conflicts. I'll update.

doronbehar commented:
There's a hash mismatch detected by ofborg. I wonder whether the difference between the hashes is the same as explained in the comment near the hash.

khaneliman committed:
python310Packages.numba: add meta.mainProgram

khaneliman commented:
> There's a hash mismatch detected by ofborg. I wonder whether the difference between the hashes is the same as explained in the comment near the hash.

Must be. Updated to latest commit from upstream while I was at it.

doronbehar commented:
@ofborg build python3.pkgs.numba

doronbehar commented:
> Must be. Updated to latest commit from upstream while I was at it.

Would it be possible to verify this please? I'm just deeply interested whether this is something funny about GitHub's `git archive` calls. You can check that by going back to the old hash. Use `diff --recursive` between that `nix build -Lf. python3.pkgs.numba.src` result and the manually downloaded and extracted tarball. The old hash, according to ofborg it was:

```
error: hash mismatch in fixed-output derivation '/nix/store/gmbmcx9w50spng382divw5ajckbpgwc7-source.drv':
         specified: sha256-38VpO2wI7IDYm96FH+93xVJeN1HA/lD66RO4svxkgUg=
            got:    sha256-oJZL0JXcmGhCm6EhGW+f3PQ86jz3g8OMIO3pnH5wQxY=
```

See previous build attempts [here](https://logs.ofborg.org/?key=nixos%2Fnixpkgs.248370).

khaneliman commented:
> > Must be. Updated to latest commit from upstream while I was at it.
> 
> 
> 
> Would it be possible to verify this please? I'm just deeply interested whether this is something funny about GitHub's `git archive` calls. You can check that by going back to the old hash. Use `diff --recursive` between that `nix build -Lf. python3.pkgs.numba.src` result and the manually downloaded and extracted tarball. The old hash, according to ofborg it was:
> 
> 
> 
> ```
> 
> error: hash mismatch in fixed-output derivation '/nix/store/gmbmcx9w50spng382divw5ajckbpgwc7-source.drv':
> 
>          specified: sha256-38VpO2wI7IDYm96FH+93xVJeN1HA/lD66RO4svxkgUg=
> 
>             got:    sha256-oJZL0JXcmGhCm6EhGW+f3PQ86jz3g8OMIO3pnH5wQxY=
> 
> ```
> 
> 
> 
> See previous build attempts [here](https://logs.ofborg.org/?key=nixos%2Fnixpkgs.248370).

I saw the same hash locally when I did a build of the previous commit after the rebase with that new logic. I just updated to latest commit since this wasn't merged yet. I won't be able to dive deep into it for a bit since im packing for vacation.
